components/backend/main.go

```diff
@@ -520,6 +534,42 @@ func parseSpec(spec map[string]interface{}) AgenticSessionSpec {
 	return result
 }
 
+// Write session data to persistent files
+func writeDataToFiles(sessionName string, statusUpdate map[string]interface{}) {
```

Comment:
Need yo update the backend to pull from the file storage too and relay that back to the UI